tgstation/tgstation
# Ports /vg/'s SQL migration controller.

## About The Pull Request

When porting BSQL to /vg/, I noticed TG maintainers need to update the database manually for each PR pertaining to SQL.

On /vg/, we have a system which automatically updates the DB whenever a PR a merged if the contributor coded a migration file, so I figured I would port it to TG now that I got it to work for BSQL.

Testing done :
- download and compile TG's source code
- add the migration controller
- set up a local DB with WAMP and use the SQL schema to make it work
- create a very basic migration file (adding an empty table)
- check if the DB is updated in phpMyAdmin, look for roundstart runtimes, see if I can join. 

I've included the DB Major/Minor table in this PR, however do note that the migration controller essentially makes it redundant, as it has its own check to see if the database is up to date or not.

## Why It's Good For The Game

Let contributors modify the database more easily. Any maintainer can now merge a PR pertaining to SQL, instead of only the maintainer with DB access.
